synchronizer/synchronizer.go

```diff
@@ -439,18 +468,20 @@ func (s *ClientSynchronizer) checkReorg(latestBlock *state.Block) (*state.Block,
 			// Reorg detected. Getting previous block
 			dbTx, err := s.state.BeginStateTransaction(s.ctx)
 			if err != nil {
-				log.Fatalf("error creating db transaction to get prevoius blocks")
+				log.Errorf("error creating db transaction to get prevoius blocks")
+				return nil, err
 			}
 			latestBlock, err = s.state.GetPreviousBlock(s.ctx, depth, dbTx)
 			errC := dbTx.Commit(s.ctx)
 			if errC != nil {
 				log.Errorf("error committing dbTx, err: %w", errC)
 				rollbackErr := dbTx.Rollback(s.ctx)
 				if rollbackErr != nil {
-					log.Fatalf("error rolling back state. RollbackErr: %w, err: %w",
-						rollbackErr, errC)
+					log.Errorf("error rolling back state. RollbackErr: %w, err: %w", rollbackErr, errC)
```

Comment:
```suggestion
					log.Errorf("error rolling back state. RollbackErr: %w", rollbackErr)
```

I don't think we need to parse the error twice